infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutKmPostDao.kt

```diff
@@ -54,6 +54,28 @@ class LayoutKmPostDao(jdbcTemplateParam: NamedParameterJdbcTemplate?)
         }
     }
 
+    fun fetchVersionsForPublication(
+        trackNumberId: IntId<TrackLayoutTrackNumber>,
+        kmPostIdsToPublish: List<IntId<TrackLayoutKmPost>>,
+    ): List<RowVersion<TrackLayoutKmPost>> {
+        val sql = """
+            select km_post.row_id, km_post.row_version
+            from layout.km_post_publication_view km_post
+            where (('DRAFT' = any(km_post.publication_states)
+                     and km_post.official_id in (:km_post_ids_to_publish))
+                   or ('OFFICIAL' = any(km_post.publication_states)
+                         and (km_post.official_id in (:km_post_ids_to_publish) is distinct from true)))
```

Comment:
Tämä `is distinct from true` tyhjille listoille on hyvä workaroundi. Tälle voisi vaan pistää tähän ja tuonne toiseen paikkaan jonkinlaisen ihan kooditason kommentin, että tällä workaroundataan nimenomaan tätä ongelmaa. Nyt tätä lukiessa se on aika selkeä kun se on kommentoitu reviewissä hyvin, mutta esim. vuoden päästä tätä lukiessa se ei välttämättä olekaan enää muistissa.